src/NHibernate/Impl/FutureValueAsync.cs

```diff
@@ -18,18 +18,13 @@ public FutureValueAsync(GetResult result)
 
 		public async Task<T> GetValue()
 		{
-			var result = await getResult();
+			var result = await getResult().ConfigureAwait(false);
 			if (ExecuteOnEval != null)
 				// When not null, ExecuteOnEval is fetched with PostExecuteTransformer from IntermediateHqlTree
 				// through ExpressionToHqlTranslationResults, which requires a IQueryable as input and directly
 				// yields the scalar result when the query is scalar.
 				return (T)ExecuteOnEval.DynamicInvoke(result.AsQueryable());
-			var enumerator = result.GetEnumerator();
-
-			if (!enumerator.MoveNext())
-				return default(T);
-
-			return enumerator.Current;
+			return result.FirstOrDefault();
```

Comment:
Looks like old code inherited from .Net Framework 1 times. This was leaking an enumerator, without disposing it. Synchronous version has the same issue and should be fixed too.